girder/girder_large_image/web_client/vue/components/PresetsMenu.vue

```diff
@@ -0,0 +1,148 @@
+<script>
+import { restRequest } from '@girder/core/rest';
+
+export default {
+    props: ['itemId', 'currentMode', 'currentFrame', 'currentStyle'],
+    emits: ['setCurrentMode', 'setCurrentFrame', 'updateStyle'],
+    data() {
+        return {
+            availablePresets: [],
+            selectedPreset: undefined,
+            showPresetCreation: false,
+            newPresetName: undefined,
+            errorMessage: undefined,
+        }
+    },
+    methods: {
+        getPresets() {
+            restRequest({
+                type: 'GET',
+                url: 'item/' + this.itemId + '/internal_metadata/presets',
+            }).then((presets) => {
+                if (presets) {
+                    this.availablePresets = presets
+                }
+            })
+        },
+        addPreset() {
+            const newPreset = {
+                'name': this.newPresetName || this.generatedPresetName,
+                'mode': this.currentMode,
+                'frame': this.currentFrame,
+                'style': this.currentStyle,
+            }
+            if (this.availablePresets.find((p) => p.name === newPreset.name)) {
```

Comment:
We should trim the name before comparison and saving it.  I made a preset called "foo" and it let me create "foo ", which looks like a name collision.